packages/entities/entities-vaults/src/components/VaultForm.vue

```diff
@@ -847,6 +849,7 @@ const updateFormValues = (data: Record<string, any>): void => {
   if (config && (Object.keys(config).length || data?.name === VaultProviders.KONNECT)) {
     vaultProvider.value = data?.item?.name || data?.name || ''
     originalVaultProvider.value = vaultProvider.value
+    configStoreId.value = data?.config_store_id || undefined
```

Comment:
I encountered an issue where the secrets were gone after updating a Konnect vault. It seems the UI is creating a new `config-store` upon `vault`'s update:

https://github.com/user-attachments/assets/ff9070ed-5071-4eaf-b907-d1ab2cf5ceb3

After some quick investigation, `config_store_id` fields appear to be nested under different hierarchies in the responses of the following endpoints:

#### /runtime_groups/0c9a728b-8cf5-450c-97c3-995876d13495/v1/vaults/87ea53c8-2eea-4056-a3e7-934d07f7e210

<img width="541" alt="Screenshot 2024-09-09 at 17 18 45" src="https://github.com/user-attachments/assets/80415b87-7297-4a7f-9ab5-49076f0bdff3">

#### /v2/control-planes/0c9a728b-8cf5-450c-97c3-995876d13495/core-entities/vaults/87ea53c8-2eea-4056-a3e7-934d07f7e210

<img width="544" alt="Screenshot 2024-09-09 at 17 20 39" src="https://github.com/user-attachments/assets/efe3d430-05d0-41dd-886c-5d008e32106b">

I doubt that made the ref `configStoreId` here null-ish while updating a vault.

Reply:
I pushed a [fix](https://github.com/Kong/public-ui-components/pull/1601/commits/a64a6e2f64c4b11e1dbad6f65013cfef4af81a64)

Reply:
I can confirm that this is fixed. Thanks